packages/client/src/hooks/useNotifications.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  nextNotification,
+  evictNotification,
+} from "@/store/actions/notificationsActions";
+import { getActiveNotification } from "@/store/selectors/notifications";
+import { NotificationInterface } from "@/types/store";
+import { useEffect, useRef } from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+interface useNotificationsParams {
+  timeouts: {
+    minTimeout?: number;
+    maxTimeout?: number;
+  };
+}
+
+interface useNotifications {
+  active: NotificationInterface;
+  handleRemoveNotification: () => void;
+}
+
+const useNotifications: (timeouts: useNotificationsParams) => {
+  active: NotificationInterface;
+  handleRemoveNotification: () => void;
+} = ({ timeouts: { minTimeout = 3000, maxTimeout = 4000 } }) => {
+  const dispatch = useDispatch();
+  const active = useSelector(getActiveNotification);
+  const handleRemoveNotification = () => {
+    dispatch(evictNotification());
+  };
+  console.log({ active });
+  const afterMinTimeout = useRef<NodeJS.Timeout | null>(null);
+  const afterMaxTimeout = useRef<NodeJS.Timeout | null>(null);
+
+  /**
+   * Possible bug: when I click next when it's exiting/other one is entering it's skipped
+   */
+  useEffect(() => {
+    afterMinTimeout.current && clearTimeout(afterMinTimeout.current);
+    afterMaxTimeout.current && clearTimeout(afterMaxTimeout.current);
```

Comment:
Ok, so this is good enough, but...this works in most cases, not in one.
It works to prevent timeouts firing against wrong notification ✅ 
It won't clear timeout on dismount:
- the timeout is set
- in the mean time (before timeout ran out) the hook is dismounted (the component calling a hook is dismounted)
- the timeout runs out and fires a callback

This shouldn't be a problem in this use case, but:
- it represents a memory leak
- would in very similar cases (this exact case, but having multiple components consuming/depending on the store) produce unexpected behaviour: the function running after it should have been cleaned up (on dismount)

Reply:
Additionally, your case could be written like so:
```typescript
useEffect(() => {
  afterMinTimeout.current = setTimeout(() => {
    dispatch(nextNotification());
  }, minTimeout);

  afterMaxTimeout.current = setTimeout(() => {
    dispatch(evictNotification());
  }, maxTimeout);

  return () => {
    afterMinTimeout.current && clearTimeout(afterMinTimeout.current);
    afterMaxTimeout.current && clearTimeout(afterMaxTimeout.current);
  };
}, [active]);
```
This way, the `useEffect` is ran at mount. When the `active` changes, it should be ran again, however, before it is ran again, the cleanup function is ran first, clearing timeout, only then is the `useEffect` block ran again, setting up new timeouts, and rinse and repeat for each change of `active`. Finally, when the hook is dismounted, it will run a cleanup function, clearing all (if any) outstanding timeouts.